core/jazz_acl/components/casbin.js

```diff
@@ -17,55 +17,185 @@
+    let filteredPolicies = result.data.filter(el => el.length >=1);
+    if (filteredPolicies.length) {
+      filteredPolicies = filteredPolicies.map(policyArr => policyArr.map(policy => [policy[0], policy[1].split('_')[1], policy[2]]));
+      result.data = filteredPolicies;
+    } else {
+      result.data = [];
+    }
+  }
+  
+  return result;
 }
 
-/* Get the policies from casbin given the value*/
+/* Get the policies from casbin given the values and index*/
 async function getFilteredPolicy(index, values, config) {
-  //TODO Implement getFilteredPolicy
+  let result = {};
+  let conn, enforcer;
+  
+  try {
+    conn = await dbConnection(config);
+    enforcer = await Enforcer.newEnforcer('./config/rbac_model.conf', conn);
+    
+    const promisedPolicies = values.map(async value => await enforcer.getFilteredPolicy(index, value));
+    const policies = await Promise.all(promisedPolicies);
+    result = {
+      "success": true,
+      "data": policies
+    };
+  } catch(err) {
+    logger.error(err.message);
+    result = {
+      "success": false,
+      "error": err.message
+    };
+  }
+  await conn.close();
   return result;
 }
 
-async function removeFilteredPolicy(index, value) {
-  //TODO: implement removeFilteredPolicy
+/* Check permissions for a user */
+async function enforce(userId, serviceId, category, permission, config) {
+  let result = {};
+  
+  try {
+    const conn = await dbConnection(config);
+    const enforcer = await Enforcer.newEnforcer('./config/rbac_model.conf', conn);
+    result.authorized = enforcer.enforce(userId, `${serviceId}_${category}`, permission);
+  } catch(err) {
+    logger.error(err.message);
+    result = {
+      authorized: false,
+      error: err.message
+    }
+  }
 
-  return true;
+  return result;
 }
 
-async function getPermissionForUser(user) {
-  //TODO: implement getPermissionForUser
+/* Remove filtered policy */
+async function removeFilteredPolicy(serviceId, config) {
+  let result = {};
+  const objects = [`${serviceId}_manage`, `${serviceId}_code`, `${serviceId}_deploy`];
+
+  try {
+    const getPolicies = await getFilteredPolicy(1, objects, config);
+    
+    if (getPolicies.data) {
+      const totalPolicies = getPolicies.data.length;  
+      let removeResult = [];
+      
+      if (totalPolicies) {
+        const conn = await dbConnection(config);
+        const enforcer = await Enforcer.newEnforcer('./config/rbac_model.conf', conn);
+        removeResult = getPolicies.data.map(async policies => {
+          return policies.map(async policy => {
+            await enforcer.removePolicy(policy[0], policy[1], policy[2]);
+          });
+        }); 
+        
+        removeResult = await Promise.all(removeResult);
+        await enforcer.savePolicy();
+        await conn.close();
+      }
+      
+      if (removeResult.length === totalPolicies) {
+        result.success = true;
+      } else {
+        result = {
+          "success": false,
```

Comment:
`"success": false` is redundant (I think it is understood, since `error` already exists in the response).